arrow/src/main/java/org/apache/iceberg/arrow/vectorized/GenericArrowVectorAccessorFactory.java

```diff
@@ -156,6 +158,8 @@ public ArrowVectorAccessor<DecimalT, Utf8StringT, ArrayT, ChildVectorT> getVecto
           return new DictionaryFloatAccessor<>((IntVector) vector, dictionary);
         case INT64:
           return new DictionaryLongAccessor<>((IntVector) vector, dictionary);
+        case INT96:
+          return new DictionaryTimestampInt96Accessor<>((IntVector) vector, dictionary);
```

Comment:
we probably want to provide some historical contexts for INT96, maybe add a comment to link to https://issues.apache.org/jira/browse/PARQUET-323